File: sample/src/main/java/ru/surfstudio/standard/interactor/common/error/UnknownServerEnum.java
package ru.surfstudio.standard.interactor.common.error;

/**
 * Ошибка если с сервера приходит неизвестный enum
 */

public class UnknownServerEnum extends IllegalArgumentException {
    private static final String ERROR_MESSAGE = "Неизвестное значение enum: ";

    public UnknownServerEnum(int value) {
        super(ERROR_MESSAGE + value);
    }

    public UnknownServerEnum(Class clazz, int value) {
        super(ERROR_MESSAGE + value + " для класса " + clazz.getName());
    }

    public UnknownServerEnum(Class clazz, String value) {
        super(ERROR_MESSAGE + value + " для класса " + clazz.getName());
    }
}

File: template/src/main/java/ru/surfstudio/standard/interactor/common/error/UnknownServerEnum.kt
package ru.surfstudio.standard.interactor.common.error

/**
 * Ошибка если с сервера приходит неизвестный enum
 */

class UnknownServerEnum : IllegalArgumentException {

    constructor(value: Int) : super(ERROR_MESSAGE + value) {}

    constructor(clazz: Class<*>, value: Int) : super(ERROR_MESSAGE + value + " для класса " + clazz.name) {}

    constructor(clazz: Class<*>, value: String) : super(ERROR_MESSAGE + value + " для класса " + clazz.name) {}

    companion object {
        private val ERROR_MESSAGE = "Неизвестное значение enum: "
    }
}
